### ISSUE TITLE
Questions about reproducing the results

### ISSUE THREAD

**Issue by SheldonWongww** [CrystalSixone/VLN-GOAT#16]
Hello, thanks for your brilliant work VLN-GOAT, but i have a question about the reproduced results. As you can see in the screenshot, after pre-training on R2R, why is the size of the resulting weight file _model_step_best.pt_ different from the one you provided on Hugging Face? Furthermore, after fine-tuning with this weight file, I've found that the SR and SPL metrics don't achieve the expected results. My reproduced results on _val_seen_ are **SR 82.08** and **SPL 76.08**, and on _val_unseen_ they are **SR 76.67** and **SPL 65.57**. 

Thank you again, and I wish you a pleasant day.

![Image](https://github.com/user-attachments/assets/be7cdcb4-6ba1-4c33-8cb1-f192b65a9da1)

**Comment by CrystalSixone** [comment]
Hi, thank you very much for your interest in our work!

Regarding your questions:
1. Checkpoint File Size
The difference in size between your reproduced model_step_best.pt and the one on Hugging Face is expected. In earlier versions of the code, some unnecessary parameters were initialized but never actually used in the forward pass. When preparing the official release, we cleaned up the code, but the released checkpoint may still contain these unused parameters. Functionally, this does not affect model behavior.

2. Reproduced Results
It's difficult to pinpoint the cause without more details about your training setup (e.g., GPU type, batch size, learning rate, number of iterations, etc.). From experience, these factors can influence final performance.

If you're willing to share more about your setup, I’d be happy to take a closer look.
Thanks again, and I hope you have a great day!

**Comment by SheldonWongww** [comment]
Thank you very much for your kindness. Regarding the experimental setup, I completed the pre-training and fine-tuning on a single NVIDIA GeForce RTX 3090. The fine-tuning process took about 5 days.

The content of the bash script _pretrain_src/run_r2r_goat.sh_ is as follows:

name=goat_r2r_pretrain
DATA_ROOT=../datasets/R2R/
NODE_RANK=0
NUM_GPUS=1

CUDA_VISIBLE_DEVICES='1' python -m torch.distributed.run \
  --nproc_per_node=${NUM_GPUS} --node_rank ${NODE_RANK} --master_port 8887 \
  train_r2r_goat.py --world_size ${NUM_GPUS} \
  --name ${name} \
  --vlnbert cmt \
  --model_config config/r2r_GOAT_model_config.json \
  --config config/r2r_GOAT_pretrain.json \
  --root_dir $DATA_ROOT \
  --cuda_first_device 0


And for _map_nav_src/scripts/run_r2r_goat.sh_:
name=goat_r2r
DATA_ROOT=../datasets
train_alg=dagger
ft_dim=768
features=clip768
ngpus=1
seed=0
outdir=${DATA_ROOT}/R2R/
aug_file=${DATA_ROOT}/R2R/annotations/prevalent_aug_train_enc.json
speaker_file=${DATA_ROOT}/R2R/speaker/transpeaker_r2r/state_dict/best_both_bleu.pt
r2r_pretrain_file=${DATA_ROOT}/R2R/pretrain/goat_r2r_pretrain/ckpts/model_step_best.pt

flag="--root_dir ${DATA_ROOT}
   --dataset r2r
   --output_dir ${outdir}
   --world_size ${ngpus}
   --seed ${seed}
   --tokenizer roberta
   --name ${name}  
   --mode train
   --enc_full_graph
   --graph_sprels
   --fusion dynamic
   --expert_policy spl
   --train_alg ${train_alg}
   
   --num_l_layers 6
   --num_x_layers 3
   --num_pano_layers 2
   
   --max_action_len 15
   --max_instr_len 200
   --batch_size 12
   --lr 2e-5
   --iters 150000
   --log_every 1000
   --optim adamW
   --features ${features}
   --image_feat_size ${ft_dim}
   --angle_feat_size 4
   --ml_weight 0.2 
   --feat_dropout 0.5
   --dropout 0.1
   --use_transpeaker
   --speaker ${speaker_file}
   --accumulateGrad
   --aug ${aug_file}
   --do_back_txt
   --do_back_img
   --do_back_txt_type type_2
   --do_back_imgobj_type type_1
   --do_add_method door
   --z_instr_update
   --do_front_txt
   --do_front_img
   --do_front_his
   "

CUDA_VISIBLE_DEVICES='1' python r2r/main_nav.py $flag  \
   --bert_ckpt_file ${r2r_pretrain_file}

If I remember correctly, I did not modify parameters such as batch size, learning rate, or number of iterations. I found that during fine-tuning, the best iteration was 27000 (or **26000** when viewed in Tensorboard).

![Image](https://github.com/user-attachments/assets/2e7de590-cc52-4995-9b16-67577b745421)

**Comment by CrystalSixone** [comment]
Hi, sorry for the late reply. That does sound unusual. I re-ran the fine-tuning using our original pre-trained file, and the results are as expected—performance peaks around iteration 77k. You might want to try fine-tuning with our released pre-trained file to see if it resolves the issue. I'm also in the process of reviewing the pre-training code to identify any differences introduced during the cleaning. Thanks again for bringing this to my attention.
